docs/abtest.rst

```diff
@@ -0,0 +1,127 @@
+.. This Source Code Form is subject to the terms of the Mozilla Public
+.. License, v. 2.0. If a copy of the MPL was not distributed with this
+.. file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+.. _ab_testing:
+
+=================
+A/B Testing
+=================
+
+A/B testing on bedrock is done with the help of `Traffic Cop
+<https://github.com/mozilla/trafficcop/>`_. A small javascript library which will
+direct site traffic to different variants in a/b experiments and make sure a
+visitor always sees the same variation.
+
+It's possible to test more than 2 variants.
+
+Traffic Cop sends users to experiments and then we use Google Analytics (GA) to
+analyze which variation is more successful. (If the user has DNT enabled they
+do not participate in experiments.)
+
+All a/b tests should have a `mana page <https://mana.mozilla.org/wiki/display/EN/Details+of+experiments+by+mozilla.org+team>`_
+detailing the experiment and recording the results.
+
+Coding the variants
+-------------------
+
+Traffic cop supports two methods of a/b testing. Executing different on page
+javascript or  redirecting to the same URL with a query string appended. We
+mostly use the redirect method in bedrock. This makes testing easier.
+
+Create a `variation view <http://bedrock.readthedocs.io/en/latest/coding.html#variation-views>`_
+for the a/b test.
+
+The view can handle the url redirect in one of two ways:
+
+1. the same page, with some different content based on the `variation` variable
+1. a totally different page
+
+Content variation
+~~~~~~~~~~~~~~~~~
+
+Useful for small focused tests.
+
+This is explained on the `variation view <http://bedrock.readthedocs.io/en/latest/coding.html#variation-views>`_
+page.
+
+New page
+~~~~~~~~
+
+Useful for large page changes where content and assets are dramatically
+different.
+
+Create the variant page like you would a new page. Make sure it is ``noindex``
+and does not have a ``canonical`` url.
+
+```
+{% block canonical_urls %}<meta name="robots" content="noindex,follow">{% endblock %}
+```
+
+Configure as explained on the `variation view <http://bedrock.readthedocs.io/en/latest/coding.html#variation-views>`_
+page.
+
+Traffic Cop
+-----------
+
+Create a .js file where you initialize Traffic Cop and include that in the
+experiments block in the template that will be doing the redirection. Wrap the
+extra js include in a `switch <TODO>`_.
```

Comment:
Is this TODO still, or just as an example?

Reply:
Oops, it's supposed to be a link.